js/modules/k6/grpc/client.go

```diff
@@ -421,7 +526,43 @@ func (c *Client) parseConnectParams(raw map[string]interface{}) (connectParams,
 			if params.MaxSendSize < 0 {
 				return params, fmt.Errorf("invalid maxSendSize value: '%#v, it needs to be a positive integer", v)
 			}
+		case "tls":
+			var ok bool
+			params.TLS, ok = v.(map[string]interface{})
 
+			if !ok {
+				return params, fmt.Errorf("invalid tls value: '%#v', expected (optional) keys: cert, key, password, and cacerts", v)
+			}
+			// optional map keys below
+			if cert, certok := params.TLS["cert"]; certok {
+				if _, ok = cert.(string); !ok {
+					return params, fmt.Errorf("invalid tls cert value: '%#v', it needs to be a PEM formatted string", v)
+				}
+			}
+			if key, keyok := params.TLS["key"]; keyok {
+				if _, ok = key.(string); !ok {
+					return params, fmt.Errorf("invalid tls key value: '%#v', it needs to be a PEM formatted string", v)
+				}
+			}
+			if pass, passok := params.TLS["password"]; passok {
+				if _, ok = pass.(string); !ok {
+					return params, fmt.Errorf("invalid tls password value: '%#v', it needs to be a string", v)
+				}
+			}
+			if cacerts, cacertsok := params.TLS["cacerts"]; cacertsok {
+				var cacertsArray []interface{}
+				if cacertsArray, ok = cacerts.([]interface{}); ok {
+					for _, cacertsArrayEntry := range cacertsArray {
+						if _, ok = cacertsArrayEntry.(string); !ok {
+							return params, fmt.Errorf("invalid tls cacerts value: '%#v',"+
+								" it needs to be a string or string[] of PEM formatted strings", v)
```

Comment:
Just to make this more human-oriented :relaxed: feel free to ignore
```suggestion
								" it needs to be a string or an array of PEM formatted strings", v)
```